hera_sim/tests/test_vis.py

```diff
@@ -339,429 +459,189 @@ def align_src_to_healpix(point_source_pos, point_source_flux, nside=2 ** 4):
-        point_source_pos, point_source_flux
-    )
-
-    viscpu = VisCPU(
-        uvdata=uvdata2,
-        sky_freqs=freqs,
-        point_source_flux=point_source_flux,
-        point_source_pos=point_source_pos,
-        nside=2 ** 4,
-    ).simulate()
+    viscpu = VisibilitySimulation(data_model=model_data, simulator=cpu).simulate()
     viscpu *= 2.0  # account for factor of 2 between Stokes I and 'xx' pol.
 
-    healvis = HealVis(
-        uvdata=uvdata2,
-        sky_freqs=freqs,
-        point_source_flux=point_source_flux,
-        point_source_pos=point_source_pos,
-        nside=2 ** 4,
+    healvis = VisibilitySimulation(
+        data_model=model_data, simulator=healvis, n_side=2 ** 4
     ).simulate()
 
     assert viscpu.shape == healvis.shape
-    np.testing.assert_allclose(viscpu.conj(), healvis, rtol=0.05)
+    np.testing.assert_allclose(viscpu, healvis, rtol=0.05)
 
 
-def test_comparision_horizon(uvdata2):
-    freqs = np.unique(uvdata2.freq_array)
+def test_vis_cpu_pol_gpu(uvdata_linear):
+    old = vis_cpu.HAVE_GPU
 
-    # put a point source in
-    point_source_pos = np.array(
-        [[0, uvdata2.telescope_location_lat_lon_alt[0] + np.pi / 2]]
-    )
-    point_source_flux = np.array([[1.0]] * len(freqs))
+    vis_cpu.HAVE_GPU = True
 
-    # align to healpix center for direct comparision
-    point_source_pos, point_source_flux = align_src_to_healpix(
-        point_source_pos, point_source_flux
+    uvdata_linear.polarization_array = [-8, -7, -6, -5]
+    beam = PolyBeam(polarized=True)
+
+    sky_model = make_point_sky(
+        uvdata_linear,
+        ra=np.linspace(0, 2 * np.pi, 8) * rad,
+        dec=uvdata_linear.telescope_location_lat_lon_alt[0] * np.ones(8) * rad,
+        align=False,
     )
 
-    viscpu = VisCPU(
-        uvdata=uvdata2,
-        sky_freqs=freqs,
-        point_source_flux=point_source_flux,
-        point_source_pos=point_source_pos,
-        nside=2 ** 4,
-    ).simulate()
-    viscpu *= 2.0  # account for factor of 2 between Stokes I and 'xx' pol.
+    simulator = VisCPU(use_gpu=True)
 
-    healvis = HealVis(
-        uvdata=uvdata2,
-        sky_freqs=freqs,
-        point_source_flux=point_source_flux,
-        point_source_pos=point_source_pos,
-        nside=2 ** 4,
-    ).simulate()
+    with pytest.raises(RuntimeError):
+        VisibilitySimulation(
+            data_model=ModelData(
+                uvdata=uvdata_linear, sky_model=sky_model, beams=[beam]
+            ),
+            simulator=simulator,
+            n_side=2 ** 4,
+        )
 
-    assert viscpu.shape == healvis.shape
-    np.testing.assert_allclose(viscpu.conj(), healvis, rtol=0.05)
+    vis_cpu.HAVE_GPU = old
 
 
-def test_comparison_multiple(uvdata2):
-    freqs = np.unique(uvdata2.freq_array)
+@pytest.mark.parametrize(
+    "simulator, order, conj",
+    product(
+        SIMULATORS, ["time", "baseline", "ant1", "ant2"], ["ant1<ant2", "ant2<ant1"]
+    ),
```

Comment:
Does this take the outer product of the inputs?

Reply:
Yes.

Reply:
Neat. Is this syntax preferred to just using multiple `@pytest.parametrize` decorators? If so, then there are some test functions I've written that I should update in the future.